pkg/parquetquery/predicates.go

```diff
@@ -478,3 +478,94 @@ func (m *CallbackPredicate) KeepColumnChunk(*ColumnChunkHelper) bool { return m.
 func (m *CallbackPredicate) KeepPage(pq.Page) bool { return m.cb() }
 
 func (m *CallbackPredicate) KeepValue(pq.Value) bool { return m.cb() }
+
+var _ Predicate = (*NilKeyPredicate)(nil)
+
+type NilKeyPredicate struct {
```

Comment:
It seems `NilKeyPredicate` has the same `KeepValue` logic as the `ByteEqualPredicate`, but omits the logic skipping column chunks and pages that don't contain the value. Why isn't it possible to use `ByteEqualPredicate` instead of `NilKeyPredicate`